### PR TITLE
I cannot find any use for this code.

### PR DESCRIPTION
**Pull request by guy-maurel**
I think, this is dead code.

### CODE REVIEW

File: src/defines.cpp [hunk 1]
```diff
@@ -1,117 +0,0 @@
-/**
- * @file defines.cpp
- * Manages the table of defines for some future time when these will be used to
- * help decide whether a block of #if'd code should be formatted.
- *
- * !! This isn't used right now. !!
```
Comment (mwoehlke-kitware) [inline src/defines.cpp]: Well, this comment would seem to agree with your belief :slightly_smiling_face:.

File: src/option.cpp [hunk 2]
```diff
@@ -2297,10 +2296,6 @@ void process_option_line(char *configLine, const char *filename)
          add_keyword(args[idx], CT_TYPE);
       }
    }
-   else if (strcasecmp(args[0], "define") == 0)
```
Comment (mwoehlke-kitware) [inline src/option.cpp]: Hmm...we *could* have this omit a more specific warning (and get rid of it fully in a few releases)... up to you though.

File: src/defines.cpp [hunk 3]
```diff
@@ -1,117 +0,0 @@
...
-#include <cstdlib>
-#include <map>
-#include "unc_ctype.h"
-#include "chunk_list.h"
-#include "prototypes.h"
-
-using namespace std;
-
-typedef map<string, string> defmap;
-defmap defines;
```
Comment (mwoehlke-kitware) [inline src/defines.cpp]: Hmm... I don't see any way this could even be accessed outside this file. That's pretty good evidence of non-usefulness.

**Comment by mihaipopescu** [comment]
In theory, I think this could be a good feature.

I guess the intention was to be able to evaluate some `#if`s while formatting the code. Right now, uncrustify has poor support over this like any other formatting tool. If you would check `clang-format` (but also `cppclean`) source code you would see that they are using a basic define evaluation strategy. Basically what's obviously true or false they take that path (like `#if 0`) otherwise they always take the first path of the condition. At the same time, clang tooling usually works with a compilation database, a json file that tells clang how to compile certain files, including the defines. Well, it's easy for clang-format as it's using the same clang lib tooling in the back but for uncrustify would mean a great deal.

Right now, there are some cases in which uncrustify has issues formatting code around defines (disclaimer publicly written in our wiki).

```cpp
#if MY_DEFINE
void foo(char *arg)
#else
void foo(wchar_t* arg)
#endif
{
    print(arg);
}
```

A lot of tools have issues parsing the snippet above. We have also have turned this in our advantage saying that more often than not, these are examples of a code smell and can be factored to better, more comprehensible forms.

**Comment by gmaurel** [comment]
@mihaipopescu our tests data has not any test which use the functions defined in `defines.cpp`.
`defines.cpp` don't provide any function such as `get` to do anything with the stored data.
It is not sure the defines.cpp is related to the preprocessor. Only 7 characters "defines" are similar to
a preprocessor statement.
We don't have any information about the data one can store, only two strings may be provided: "abc", "1234", or "+-%", which  use?

**Pull request merged**